core/services/headreporter/telemetry_reporter_test.go

```diff
@@ -0,0 +1,144 @@
+package headreporter_test
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/protobuf/proto"
+
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	ubig "github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils/big"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/legacyevm"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/legacyevm/mocks"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/services/headreporter"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	mocks2 "github.com/smartcontractkit/chainlink/v2/core/services/telemetry/mocks"
+)
+
+type IngressAgent struct {
+	mock.Mock
+}
+
+func (t *IngressAgent) SendLog(telemetry []byte) {
+	_ = t.Called(telemetry)
+}
+
+func NewIngressAgent(t interface {
+	mock.TestingT
+	Cleanup(func())
+}) *IngressAgent {
+	m := &IngressAgent{}
+	m.Mock.Test(t)
+
+	t.Cleanup(func() { m.AssertExpectations(t) })
+
+	return m
+}
```

Comment:
Where did this code come from? It looks like a generated mock. Should it be a proper mock in a separate file.

Reply:
fixed